### ISSUE TITLE
[1.0-rc1] GROUP BY returns multiple groups for the same tag

### ISSUE THREAD

**Issue by dsw6** [influxdata/influxdb#7220]
System Info:
Influx 1.0-rc1
Ubuntu 14.04.4 

The GROUP BY command is returning multiple groups for the same tag in the result set.  This appears to happen when the query's time filter spans two days (crosses midnight).

In the example below, the results show two distant groups for each tag:  group1: results before midnight, group2: results after midnight.

I'm using Grafana to graph the data and this results in the graph showing two distinct groups on the timeline.  The before midnight group and the after midnight group, each named the same, but displayed in different colors.  

Query Example:
===============

```
SELECT count(respTime) FROM "esf_events" WHERE (serviceKey='CENTRAL_MESSAGING' or serviceKey='FACILITY_SERVICE') and time > 1472167800s and time < 1472171400s GROUP BY time(10m), serviceKey  fill(0)
```

Query Results:
================
```
{
    "results": [{
        "series": [{
            "name": "esf_events",
            "tags": {
                "serviceKey": "CENTRAL_MESSAGING"
            },
            "columns": ["time", "count"],
            "values": [
                ["2016-08-25T23:30:00Z", 1],
                ["2016-08-25T23:40:00Z", 9],
                ["2016-08-25T23:50:00Z", 17],
                ["2016-08-26T00:00:00Z", 0],
                ["2016-08-26T00:10:00Z", 0],
                ["2016-08-26T00:20:00Z", 0]
            ]
        }, {
            "name": "esf_events",
            "tags": {
                "serviceKey": "FACILITY_SERVICE"
            },
            "columns": ["time", "count"],
            "values": [
                ["2016-08-25T23:30:00Z", 42],
                ["2016-08-25T23:40:00Z", 59],
                ["2016-08-25T23:50:00Z", 37],
                ["2016-08-26T00:00:00Z", 0],
                ["2016-08-26T00:10:00Z", 0],
                ["2016-08-26T00:20:00Z", 0]
            ]
        }, {
            "name": "esf_events",
            "tags": {
                "serviceKey": "CENTRAL_MESSAGING"
            },
            "columns": ["time", "count"],
            "values": [
                ["2016-08-25T23:30:00Z", 0],
                ["2016-08-25T23:40:00Z", 0],
                ["2016-08-25T23:50:00Z", 0],
                ["2016-08-26T00:00:00Z", 25],
                ["2016-08-26T00:10:00Z", 39],
                ["2016-08-26T00:20:00Z", 25]
            ]
        }, {
            "name": "esf_events",
            "tags": {
                "serviceKey": "FACILITY_SERVICE"
            },
            "columns": ["time", "count"],
            "values": [
                ["2016-08-25T23:30:00Z", 0],
                ["2016-08-25T23:40:00Z", 0],
                ["2016-08-25T23:50:00Z", 0],
                ["2016-08-26T00:00:00Z", 34],
                ["2016-08-26T00:10:00Z", 44],
                ["2016-08-26T00:20:00Z", 55]
            ]
        }]
    }]
}
```

**Comment by jsternberg** [comment]
Might be the same as #7214.

**Comment by jsternberg** [comment]
Can you give me the output of `SHOW SHARDS`?

**Comment by jsternberg** [comment]
Reproducer:

```
> CREATE DATABASE mydb WITH SHARD DURATION 6h
> INSERT cpu,host=server01 value=2 1472230800000000000
> INSERT cpu,host=server02 value=2 1472230800000000000
> INSERT cpu,host=server01 value=3 1472234400000000000
> INSERT cpu,host=server02 value=3 1472234400000000000
> SELECT mean(value) FROM cpu WHERE time >= 1472230800000000000 AND time <= 1472234400000000000 GROUP BY time(1h), host
name: cpu
tags: host=server01
time                    mean
----                    ----
1472230800000000000     2
1472234400000000000

name: cpu
tags: host=server02
time                    mean
----                    ----
1472230800000000000     2
1472234400000000000

name: cpu
tags: host=server01
time                    mean
----                    ----
1472230800000000000
1472234400000000000     3

name: cpu
tags: host=server02
time                    mean
----                    ----
1472230800000000000
1472234400000000000     3
```

Output of `SHOW SHARDS`:

```
name: mydb
----------
id      database        retention_policy        shard_group     start_time              end_time                expiry_time             owners
19      mydb            autogen                 19              2016-08-26T12:00:00Z    2016-08-26T18:00:00Z    2016-08-26T18:00:00Z
20      mydb            autogen                 20              2016-08-26T18:00:00Z    2016-08-27T00:00:00Z    2016-08-27T00:00:00Z
```

I just need you to confirm that your use case looks similar so I know I'm fixing the correct problem.

**Comment by ikovalyov** [comment]
Shards for affected db: 
```
nodes
id	database	retention_policy	shard_group	start_time	end_time	expiry_time	owners
86	"nodes"	"default"	86	"2016-07-25T00:00:00Z"	"2016-08-01T00:00:00Z"	"2016-08-01T00:00:00Z"	""
5	"nodes"	"default"	5	"2016-08-08T00:00:00Z"	"2016-08-15T00:00:00Z"	"2016-08-15T00:00:00Z"	""
75	"nodes"	"default"	75	"2016-08-15T00:00:00Z"	"2016-08-22T00:00:00Z"	"2016-08-22T00:00:00Z"	""
83	"nodes"	"default"	83	"2016-08-22T00:00:00Z"	"2016-08-29T00:00:00Z"	"2016-08-29T00:00:00Z"	""
```

**Comment by jsternberg** [comment]
Hm... strange. The shard groups for yours don't seem to correspond to different shards.

**Comment by jsternberg** [comment]
I'm not sure if your issue is different than the one I am reproducing, but my bisect seems to have 6c7d56d as the problem commit. Is there any chance you can try this with 1.0.0-beta3 and see if the problem still exists? 1.0.0-beta3 did not have that commit, but rc1 does.

**Comment by dsw6** [comment]
Show Shards output:
------------------------
            "name": "esf",
            "columns": ["id", "database", "retention_policy", "shard_group", "start_time", "end_time", "expiry_time", "owners"],
            "values": [
                [271, "esf", "rp_7d", 271, "2016-08-23T00:00:00Z", "2016-08-24T00:00:00Z", "2016-08-31T00:00:00Z", ""],
                [275, "esf", "rp_7d", 275, "2016-08-24T00:00:00Z", "2016-08-25T00:00:00Z", "2016-09-01T00:00:00Z", ""],
                [279, "esf", "rp_7d", 279, "2016-08-25T00:00:00Z", "2016-08-26T00:00:00Z", "2016-09-02T00:00:00Z", ""],
                [283, "esf", "rp_7d", 283, "2016-08-26T00:00:00Z", "2016-08-27T00:00:00Z", "2016-09-03T00:00:00Z", ""],
                [272, "esf", "rp_30d", 272, "2016-08-23T00:00:00Z", "2016-08-24T00:00:00Z", "2016-09-23T00:00:00Z", ""],
                [276, "esf", "rp_30d", 276, "2016-08-24T00:00:00Z", "2016-08-25T00:00:00Z", "2016-09-24T00:00:00Z", ""],
                [280, "esf", "rp_30d", 280, "2016-08-25T00:00:00Z", "2016-08-26T00:00:00Z", "2016-09-25T00:00:00Z", ""],
                [284, "esf", "rp_30d", 284, "2016-08-26T00:00:00Z", "2016-08-27T00:00:00Z", "2016-09-26T00:00:00Z", ""],
                [165, "esf", "rp_60d", 165, "2016-06-27T00:00:00Z", "2016-06-28T00:00:00Z", "2016-08-27T00:00:00Z", ""],
                [169, "esf", "rp_60d", 169, "2016-06-28T00:00:00Z", "2016-06-29T00:00:00Z", "2016-08-28T00:00:00Z", ""],
                [173, "esf", "rp_60d", 173, "2016-06-29T00:00:00Z", "2016-06-30T00:00:00Z", "2016-08-29T00:00:00Z", ""],
                [177, "esf", "rp_60d", 177, "2016-06-30T00:00:00Z", "2016-07-01T00:00:00Z", "2016-08-30T00:00:00Z", ""],
                [181, "esf", "rp_60d", 181, "2016-07-01T00:00:00Z", "2016-07-02T00:00:00Z", "2016-08-31T00:00:00Z", ""],
                [185, "esf", "rp_60d", 185, "2016-07-02T00:00:00Z", "2016-07-03T00:00:00Z", "2016-09-01T00:00:00Z", ""],
                [189, "esf", "rp_60d", 189, "2016-07-03T00:00:00Z", "2016-07-04T00:00:00Z", "2016-09-02T00:00:00Z", ""],
                [193, "esf", "rp_60d", 193, "2016-07-04T00:00:00Z", "2016-07-05T00:00:00Z", "2016-09-03T00:00:00Z", ""],
                [197, "esf", "rp_60d", 197, "2016-07-05T00:00:00Z", "2016-07-06T00:00:00Z", "2016-09-04T00:00:00Z", ""],
                [201, "esf", "rp_60d", 201, "2016-07-06T00:00:00Z", "2016-07-07T00:00:00Z", "2016-09-05T00:00:00Z", ""],
                [205, "esf", "rp_60d", 205, "2016-07-07T00:00:00Z", "2016-07-08T00:00:00Z", "2016-09-06T00:00:00Z", ""],
                [209, "esf", "rp_60d", 209, "2016-07-08T00:00:00Z", "2016-07-09T00:00:00Z", "2016-09-07T00:00:00Z", ""],
                [213, "esf", "rp_60d", 213, "2016-07-09T00:00:00Z", "2016-07-10T00:00:00Z", "2016-09-08T00:00:00Z", ""],
                [217, "esf", "rp_60d", 217, "2016-07-10T00:00:00Z", "2016-07-11T00:00:00Z", "2016-09-09T00:00:00Z", ""],
                [221, "esf", "rp_60d", 221, "2016-07-11T00:00:00Z", "2016-07-12T00:00:00Z", "2016-09-10T00:00:00Z", ""],
                [225, "esf", "rp_60d", 225, "2016-07-12T00:00:00Z", "2016-07-13T00:00:00Z", "2016-09-11T00:00:00Z", ""],
                [229, "esf", "rp_60d", 229, "2016-07-13T00:00:00Z", "2016-07-14T00:00:00Z", "2016-09-12T00:00:00Z", ""],
                [233, "esf", "rp_60d", 233, "2016-07-14T00:00:00Z", "2016-07-15T00:00:00Z", "2016-09-13T00:00:00Z", ""],
                [237, "esf", "rp_60d", 237, "2016-07-15T00:00:00Z", "2016-07-16T00:00:00Z", "2016-09-14T00:00:00Z", ""],
                [241, "esf", "rp_60d", 241, "2016-07-16T00:00:00Z", "2016-07-17T00:00:00Z", "2016-09-15T00:00:00Z", ""],
                [245, "esf", "rp_60d", 245, "2016-07-17T00:00:00Z", "2016-07-18T00:00:00Z", "2016-09-16T00:00:00Z", ""],
                [249, "esf", "rp_60d", 249, "2016-07-18T00:00:00Z", "2016-07-19T00:00:00Z", "2016-09-17T00:00:00Z", ""],
                [253, "esf", "rp_60d", 253, "2016-07-19T00:00:00Z", "2016-07-20T00:00:00Z", "2016-09-18T00:00:00Z", ""],
                [257, "esf", "rp_60d", 257, "2016-07-20T00:00:00Z", "2016-07-21T00:00:00Z", "2016-09-19T00:00:00Z", ""],
                [261, "esf", "rp_60d", 261, "2016-07-21T00:00:00Z", "2016-07-22T00:00:00Z", "2016-09-20T00:00:00Z", ""],
                [265, "esf", "rp_60d", 265, "2016-07-22T00:00:00Z", "2016-07-23T00:00:00Z", "2016-09-21T00:00:00Z", ""],
                [269, "esf", "rp_60d", 269, "2016-07-23T00:00:00Z", "2016-07-24T00:00:00Z", "2016-09-22T00:00:00Z", ""],
                [273, "esf", "rp_60d", 273, "2016-08-23T00:00:00Z", "2016-08-24T00:00:00Z", "2016-10-23T00:00:00Z", ""],
                [277, "esf", "rp_60d", 277, "2016-08-24T00:00:00Z", "2016-08-25T00:00:00Z", "2016-10-24T00:00:00Z", ""],
                [281, "esf", "rp_60d", 281, "2016-08-25T00:00:00Z", "2016-08-26T00:00:00Z", "2016-10-25T00:00:00Z", ""],
                [285, "esf", "rp_60d", 285, "2016-08-26T00:00:00Z", "2016-08-27T00:00:00Z", "2016-10-26T00:00:00Z", ""]
            ]

**Comment by jsternberg** [comment]
I messed up and looked at @ikovalyou's `SHOW SHARDS` output and looked at @dsw6's `SELECT` output. When I put both of your `SHOW SHARDS` output with the output of the `SELECT` you both gave me in each of your issues, it matches my reproducer.

Thanks for the help in finding this! I'll start working on a fix.

**Comment by dsw6** [comment]
I rolled by to 1.0.0-beta3 and do not see the problem.

**Comment by dswarbrick** [comment]
I can also confirm this bug. 1.0-beta3 was fine, but 1.0-rc1 exhibits the repeated groups.

**Comment by jsternberg** [comment]
We reverted the underlying commit in #7236. We will try to accomplish the same goal for 1.1, but we couldn't come up with a suitable fix in time that we felt would be safe for 1.0.
